eclipse-collections/src/main/java/com/fasterxml/jackson/datatype/eclipsecollections/EclipseCollectionsDeserializers.java

```diff
@@ -156,9 +155,11 @@
     // are faster to construct.
 
     // initialized below
-    static final Map<Class<? extends PrimitiveIterable>, JsonDeserializer<?>> PRIMITIVE_DESERIALIZERS = new HashMap<>();
+    static final Map<Class<? extends PrimitiveIterable>, JsonDeserializer<?>> PRIMITIVE_DESERIALIZERS
+            = new IdentityHashMap<>();
```

Comment:
Is there a specific reason to use `IdentityHashMap` here?

Reply:
IdentityHashMap should be a little faster in this case, which may not be a totally pointless optimization because getting serializers/deserializers by class is an operation on the hot path. I use IdentityHashMap in new code (`PrimitivePrimitiveMapSerializers` - in this PR, and in the (yet unchecked) fastutil module which should come in a follow-up PR), so I aligned existing code to use IdentityHashMap for consistency, too.

Reply:
It'd be good to measure this. I do remember `Class<?>` lookup being on hot path because it does not override default `equals()` or `hashCode()`, but I also recall `IdentityHashMap` itself not being super efficient. What core databind does is actually uses `ClassKey` type, but while I think that removed a hot spot I don't think I ever properly verified its effect with JMH.

Having said all of that, I do not think this lookup should be on critical path as `DeserializerCache` used by `jackson-databind` should handle caching by type and this provider should not be called more than once.